metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -1064,15 +1103,11 @@ def gen_dataset_workunits(
                 entityUrn=dataset_urn, aspect=global_tags
             ).as_workunit()
 
-        if (
-            isinstance(table, SnowflakeView)
-            and cast(SnowflakeView, table).view_definition is not None
-        ):
-            view = cast(SnowflakeView, table)
+        if isinstance(table, SnowflakeView) and table.view_definition is not None:
             view_properties_aspect = ViewProperties(
-                materialized=False,
+                materialized=table.materialized,
                 viewLanguage="SQL",
-                viewLogic=view.view_definition,
+                viewLogic=table.view_definition,
```

Comment:
honestly might be clearer to keep the `view = table` line - the cast is not needed though

Reply:
Eh, we refer to views as tables elsewhere in the code too, basically anywhere we do `table: Union[SnowflakeTable, SnowflakeView]` so I think it's ok with that context